impl.go

```diff
@@ -262,13 +276,14 @@ func funcs(iface string, srcDir string) ([]Func, error) {
 			continue
 		}
 
-		fn := p.funcsig(fndecl)
+		fn := p.funcsig(fndecl, spec.CommentMap.Filter(fndecl))
 		fns = append(fns, fn)
 	}
 	return fns, nil
 }
 
-const stub = "func ({{.Recv}}) {{.Name}}" +
+const stub = "{{if .Comments}}{{.Comments}}{{end}}" +
```

Comment:
How hard would it be to add an end-to-end test so that mistakes here can get caught?

Reply:
Should not be a big deal, I am happy to work on that along with other suggestions to modify current tests.

Reply:
Great!

Reply:
Similar here, I will add more tests for template generation.